Add blog post about localizing websites (#60)
openwebdocs/owd-website: pages/content/posts/localizing-your-website.md

---
author: "Vinyl Da.i'gyu-Kazotetsu"
title: "Why it is important to localize your website"
date: "2022-12-14"
tags: ["development"]
ShowToc: false
ShowBreadCrumbs: false
excerpt: "A client of mine had recently asked me why it was so important to localize their web app. They asked me, why spend money on localization when you can do it for free with Google Translate? There is so much to unpack about why that approach and mindset is wrong."
---

A client of mine had recently asked me why it was so important to localize their web app.  They told me: “we just plug Google Translate in and it’s been fine. Why spend hundreds, thousands, tens of thousands on localization when you can do it for free with Google Translate?”

There is _so_ much to unpack about why that approach and mindset is _wrong_.

> There are many different categories to consider when it comes to inclusivity of diverse audiences, whether it’s to accommodate sensory impairments, theming (dark/light mode), localization and more.  When designing your website/software, or perhaps updating it, one should consider every aspect of inclusion every step of the way to engage as wide of an audience as possible.  It is important that no portion is overlooked either, as then you risk your product appearing broken or incomplete.

Author: ```suggestion
There are many considerations when it comes to being inclusive of diverse audiences, whether it’s to accommodate sensory impairments, theming (dark/light mode), localization, and more. When designing your website or software, or even just updating it, it's important to consider all aspects of inclusion every step of the way to engage as wide of an audience as possible. It is important not to overlook anything either, as then you risk your product appearing broken or incomplete.
```

> Take alt. text attributes for images, for example.  Let’s say you have ten images, and you have properly and accurately described nine of the images using alt. text attributes (or marked them as decorative, but either way, you have followed accessibility standards for 90% of your images).  A visually impaired user that needs these attributes then goes to the page containing the image without an alt. text attribute, and suddenly, an entire element cannot be described to them.  Is it just decoration, or does it contain critical information that is essential to the rest of the page?  The screen reader doesn’t know (or maybe it could, but that’s a very big “maybe”), and the user doesn’t know.  To the visually impaired user, the web page would simply be broken.

Author: ```suggestion
Take the `alt` attribute text for images as an example.  Let’s say you have ten images, and you have properly and accurately described nine of the images by including appropriate `alt` text values (or marked them as decorative, but either way, you have followed accessibility standards for 90% of your images).  If a user who benefits from and relies upon image descriptions, such as users with cognitive differences and visually impaired screen reader users, then goes to the page containing images without `alt` text, then suddenly, the element, important content, cannot be accessed.
	
Is the image just decoration? Does the image contain critical information that is essential to the rest of the page?  The screen reader has no way of knowing and the user doesn’t know.  To the visually impaired user, the web page page is incomplete; broken even.
```

> The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as Google Translate) that is able to translate entire documents, webpages, and even verbal conversations, and they’re only going to get more powerful.  However, machine translations can only do so much: the ones that are readily available to the common folk cannot comprehend context, including user interface options or special phrases (product names, etc.).  Additionally, machine translation is not able to capture the nuances and subtleties of languages.  Often times, this results in the wrong word selection, translations of phrases that shouldn’t be translated, inconsistency in word selection throughout paragraphs, and even completely wrong translations.  (I ran into this when attempting to automatically translate a Russian website to English, and the word selection was so bad, I had to ask my Russian translator to translate it for me instead.)

Reviewer: ```suggestion
The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.  

Machine translations are not fully capable of capturing the nuances and subtleties of languages; they can translate but can't localize. Translations often result in the wrong word selection, translations of phrases that shouldn’t be translated (hot dog is "hot dog" everywhere; no one wants to eat a "_<span lang="fr-fr">chien chaud</span>_"), inconsistency in word selection throughout paragraphs, and even completely wrong translations.  
```

I removed the Russian translation part, because that is the point of this whole article, not a specific example for this section. I added the french hot dog instead. Feel free to replace my hot dog example with an example from the Russian translation.... that;s why I added it: provided a suggested template for the edit

Author: The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/)) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.

Machine translations are not fully capable of capturing the nuances and subtleties of languages; they can translate but can't localize. Translations often result in the wrong word selection, translations of phrases that shouldn’t be translated (hot dog is "hot dog" everywhere; no one wants to eat a "_<span lang="fr-fr">chien chaud</span>_"), inconsistency in word selection throughout paragraphs, and even completely wrong translations.

> Going back to our previous example of missing alt. text attributes, let’s say that you translate a portion of your website, but don’t translate everything.  Perhaps you’ve translated the content but not the interface, or vice versa, or perhaps you’ve only translated part of the content.  Now the user is viewing content on your page in two different languages, but expects to only see one.  Machine translation software is now out of the question, as it is likely to end up picking up already-translated words and attempt to “translate” them again, or be unable to pick up what the original language is.  “What happened?” the user will think. “Is the website broken?  I can’t understand half of this…I guess I should search elsewhere.”

Reviewer: ```suggestion
Just as in our previous example of the missing `alt` attributes, if you provide manual translations for a portion of your website but not everything, again a section of your content is not accessible.  Perhaps you’ve translated the content but not the interface, or vice versa, or perhaps you’ve only translated part of the content.  The user is seeing content on your page in two different languages but expects to only see one. When sections of your site are manually translated without providing internationalization features, such as the `lang` attribute on the HTML element or on translated content, machine translation software becomes out of the question. Automated translation services are likely to pick up already-translated words and attempt to “translate” them again, or be unable to determine what the original language is.  “What happened?” the user will think. “Is the website broken?  I can’t understand half of this…I'll look elsewhere.”
```

> And that’s the real kicker.  Even if the user understands the original language, they would still think that something is wrong, which would provide a negative user experience and give your competitors an opportunity to gain your lost consumers.  However, if a user is requesting an alternate language, chances are high they can’t understand the original language in the first place!  By leaving your product partially translated (or realistically, untranslated at all), you are essentially telling the users that speak other languages: “we really don’t care about you.”

Reviewer: ```suggestion
And that’s the real kicker.  Even if the user understands the original language, they would still think that something is wrong, which would provide a negative user experience and give your competitors an opportunity to gain your lost consumers.  However, if a user is requesting an alternate language, chances are high that they can’t understand the original language in the first place!  By leaving your product partially translated (or realistically, untranslated at all), you are essentially telling the users that speak other languages: “we don’t really care about you.”
```

Another benefit of localizing your website is that it can improve search engine optimization (SEO).  Search engines are more likely to rank websites that are localized for different languages higher in search results, as they are seen as providing a better user experience for users who speak those languages, the same way that websites with accessibility considerations are ranked higher.  By localizing a website, you can increase your visibility in search results and have a greater chance at gaining more organic traffic.

> Of course, it is also important to _maintain_ your localizations, rather than performing them once and calling it good.  As you update your website and add or modify content, it's important to ensure that the localized versions are kept up-to-date and accurately reflect the latest content.  This can help prevent users from coming across outdated or inaccurate information, which leads to a negative user experience and potentially drives users away from the website.

Reviewer: ```suggestion
Of course, it is also important to _maintain_ your localizations, rather than performing them once and forgetting about them.  As you update your website by adding or modifying content, it's important to ensure the localized versions are also kept up-to-date and always accurately reflect the latest content.  This helps prevent users from encountering outdated or inaccurate information, which leads to a negative user experience and potentially drives users away from the website.
```

> So, there you have it, Unnamed Client.  That is my answer to your question.

Reviewer: ```suggestion
So, there you have it, Unnamed Site Owner.  That is my answer to your question.
```

---
author: "Vinyl Da.i'gyu-Kazotetsu"
title: "Why it is important to localize your website"
date: "2022-12-14"
tags: ["development"]
ShowToc: false
ShowBreadCrumbs: false

> excerpt: "A client of mine had recently asked me why it was so important to localize their web app. They asked me, why spend money on localization when you can do it for free with Google Translate? There is so much to unpack about why that approach and mindset is wrong."

Reviewer: ```suggestion
excerpt: "A site owner recently asked me why it was so important to localize their web app.  Why spend hundreds, thousands, tens of thousands on localization when you can do it for free with Google Translate? Let's discuss why that approach and mindset work for your users."
```

Author: ```suggestion
excerpt: "A site owner recently asked me why it was so important to localize their web app.  Why spend hundreds, thousands, tens of thousands on localization when you can do it for free with Google Translate? Let's discuss why that approach and mindset does not work for your users."
```

Take the `alt` attribute text for images as an example.  Let’s say you have ten images, and you have properly and accurately described nine of the images by including appropriate `alt` text values (or marked them as decorative, but either way, you have followed accessibility standards for 90% of your images).  If a user who benefits from and relies upon image descriptions, such as users with cognitive differences and visually impaired screen reader users, then goes to the page containing images without `alt` text, then suddenly, the element, important content, cannot be accessed.
	
Is the image just decoration? Does the image contain critical information that is essential to the rest of the page?  The screen reader has no way of knowing and the user doesn’t know.  To the visually impaired user, the web page page is incomplete; broken even.

> The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/)) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.

Author: I don't think we should say "applications" because there are also services and APIs for machine translation accessible through other apps and websites that may not have their own app.

> The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/)) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.

Reviewer: mostly, i wanted to change the `is` to `are` and it sounded weird b/c software is singular and plural but sounds singular only

> The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/)) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.

Author: Would "solutions" instead of "software" sound reasonable?

> The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/)) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.

Reviewer: i removed the applications. Commit if ok with my suggestion

> Machine translations are not fully capable of capturing the nuances and subtleties of languages; they can translate but can't localize. Translations often result in the wrong word selection, translations of phrases that shouldn’t be translated (hot dog is "hot dog" everywhere; no one wants to eat a "_<span lang="fr-fr">chien chaud</span>_"), inconsistency in word selection throughout paragraphs, and even completely wrong translations.

Reviewer: ```suggestion
Machine translations are not fully capable of capturing the nuances and subtleties of languages; they can translate but they can't localize. Translations often result in the wrong word selection, translations of phrases that shouldn’t be translated (hot dog is "hot dog" everywhere; no one wants to eat a "_<span lang="fr-fr">chien chaud</span>_"), inconsistency in word selection throughout paragraphs, and even completely wrong translations.
```

> Just as in our previous example of the missing `alt` attributes, if you provide manual translations for a portion of your website but not everything, again a section of your content is not accessible.  Perhaps you’ve translated the content but not the interface, or vice versa, or perhaps you’ve only translated part of the content.  The user is seeing content on your page in two different languages but expects to only see one. When sections of your site are manually translated without providing internationalization features, such as the `lang` attribute on the HTML element or on translated content, machine translation software becomes out of the question. Automated translation services are likely to pick up already-translated words and attempt to “translate” them again, or be unable to determine what the original language is.  “What happened?” the user will think. “Is the website broken?  I can’t understand half of this…I'll look elsewhere.”

Reviewer: ```suggestion
Just as with the previous example of the missing `alt` attributes, if you provide manual translations for a portion of your website but not everything, portions of your content will not be accessible.  Perhaps you’ve translated the content but not the interface, or vice versa, or perhaps you’ve only translated part of the content.  The user is seeing content in two different languages but expects to only see one. 

When sections of your site are manually translated without providing internationalization features, such as the `lang` attribute on the HTML element or on translated content, machine translation software becomes out of the question. Automated translation services are likely to pick up already-translated words and attempt to “translate” them again, or be unable to determine what the original language is.  “What happened?” the user will think. “Is the website broken?  I can’t understand half of this…I'll look elsewhere.”
```

> And that’s the real kicker.  Even if the user understands the original language, they would still think that something is wrong, which would provide a negative user experience and give your competitors an opportunity to gain your lost consumers.  However, if a user is requesting an alternate language, chances are high that they can’t understand the original language in the first place!  By leaving your product partially translated (or realistically, untranslated at all), you are essentially telling the users that speak other languages: “we don’t really care about you.”

Reviewer: ```suggestion
And that’s the real kicker.  Even if the user understands the original language, they may still think something is wrong. This creates a negative user experience and gives your competitors an opportunity to gain the consumer you just lost.  

If a user is requesting an alternate language, chances are high that they can’t understand or are not literate in the original language.  By leaving your product partially translated (or realistically, not translated at all), you are essentially telling the users that speak other languages: “we don’t really care about you.”
```

> Another benefit of localizing your website is that it can improve search engine optimization (SEO).  Search engines are more likely to rank websites that are localized for different languages higher in search results, as they are seen as providing a better user experience for users who speak those languages, the same way that websites with accessibility considerations are ranked higher.  By localizing a website, you can increase your visibility in search results and have a greater chance at gaining more organic traffic.

Reviewer: ```suggestion
Another benefit of localizing your website is that it can improve search engine optimization (SEO).  Search engines are more likely to rank websites that are localized for different languages higher in search results, as they are seen as providing a better user experience for users who speak and read those languages, the same way that websites with accessibility considerations are ranked higher.  By localizing a website, you can increase your visibility in search results and have a greater chance of gaining more organic traffic.
```

> Of course, it is also important to _maintain_ your localizations, rather than performing them once and forgetting about them.  As you update your website by adding or modifying content, it's important to ensure the localized versions are also kept up-to-date and always accurately reflect the latest content.  This helps prevent users from encountering outdated or inaccurate information, which leads to a negative user experience and potentially drives users away from the website.

Reviewer: ```suggestion
Of course, it is also important to _maintain_ your localizations, rather than performing them once and forgetting about them.  As you update your website by adding or modifying content, it's important to ensure the localized versions are also kept up-to-date and always accurately reflect the latest content.  This helps prevent users from encountering outdated or inaccurate information. Providing updated, relevant, internationalized content leads to a positive user experience and builds customer loyalty for your website, product, and company.
```

Reviewer: made the last sentence positive. "Here's what you should do!"
